openpype/pipeline/colorspace.py

```diff
@@ -19,6 +22,54 @@
 
 class CashedData:
     remapping = None
+    python3compatible = None
+    config_version_data = None
+    ocio_config_colorspaces = {}
+
+
+class DeprecatedWarning(DeprecationWarning):
+    pass
+
+
+def deprecated(new_destination):
+    """Mark functions as deprecated.
+
+    It will result in a warning being emitted when the function is used.
+    """
+
+    func = None
+    if callable(new_destination):
+        func = new_destination
+        new_destination = None
+
+    def _decorator(decorated_func):
+        if new_destination is None:
+            warning_message = (
+                " Please check content of deprecated function to figure out"
+                " possible replacement."
+            )
+        else:
+            warning_message = " Please replace your usage with '{}'.".format(
+                new_destination
+            )
+
+        @functools.wraps(decorated_func)
+        def wrapper(*args, **kwargs):
+            warnings.simplefilter("always", DeprecatedWarning)
+            warnings.warn(
+                (
+                    "Call to deprecated function '{}'"
+                    "\nFunction was moved or removed.{}"
+                ).format(decorated_func.__name__, warning_message),
+                category=DeprecatedWarning,
+                stacklevel=4
+            )
+            return decorated_func(*args, **kwargs)
+        return wrapper
+
+    if func is None:
+        return _decorator
+    return _decorator(func)
```

Comment:
It's so unfortunate that basically the same logic exists in `openpype.lib` and we need to duplicate it here because `openpype.lib` already imports `openpype.pipeline.colorspace`. I feel like that somehow this `deprecation` decorator logic should be in a module within openpype that we can import from everywhere else.

Would be nice if it didn't need to duplicate the code here.

Reply:
I agree but it is not issue for this PR